src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -587,20 +617,41 @@
                 // see if we can decode, decrypt and deserialize the users token
                 // into a login session structure
                 let session = self.session_cache.decode(token, &self.session_key)?;
-
                 let status = session.status();
 
-                let mut new_auth = None;
-                if status != SessionStatus::Active {
-                    new_auth = self.try_refresh_token(&session);
-                    if new_auth.is_none() && status == SessionStatus::Expired {
-                        return Err(Error::ApiInvalidCredentials(
-                            "Session expired, please login again".to_string(),
-                        ));
-                    }
+                // Token found in cache and active; all good, do an early return
+                if status == SessionStatus::Active {
+                    return Ok(Some(ActorDef::user(session.id, session.attributes, None)));
                 }
 
-                Ok(Some(ActorDef::user(session.id, session.attributes, new_auth)))
+                let new_auth = match self.try_refresh_token(&session) {
+                    Ok(auth) => auth,
+                    Err(e) => {
+                        warn!("RFC-6749 5.2 Error Response returned: {:?}", e);
```

Comment:
Is RFC-6479 5.2 the only way that try_refresh_token() can fail?

Reply:
According to the docs of the openidconnect crate, yes. I still have to figure out where the 4.2.2.1 Error Responses turn up

Reply:
I was meaning that `try_refresh_token()` is an internal fn, not part of the openidconnect crate, and so may have its own logic and cause other errors than those produced by the openidconnect crate. Since you posted this PR I haven't had a chance to look at the actual code of `fn try_refresh_token()`.